### PR TITLE
Use _typeshed.Self with __enter__

### PR DESCRIPTION
**Pull request by gruebel**
Relates to #5676

I also removed `TypeVar` usage, where it became obsolete. In `cProfile` it also made sense to replace the `_SelfT` references with `Self`.

### CODE REVIEW

File: stdlib/bz2.pyi [hunk 1]
```diff
@@ -82,7 +82,7 @@ def open(
 ) -> TextIO: ...
 
 class BZ2File(BaseStream, IO[bytes]):
-    def __enter__(self: _T) -> _T: ...
+    def __enter__(self: Self) -> Self: ...
```
Comment (JelleZijlstra) [inline stdlib/bz2.pyi:85]: This method doesn't actually exist at runtime; I guess it is inherited from an IO-related base class.

**Review by JelleZijlstra — APPROVED** [review]
Thanks! I double-checked all these except the Python 2 stuff.

File: stdlib/bz2.pyi (hunk 1 above)
Comment (gruebel) [reply stdlib/bz2.pyi:85]: You are correct, I think it is IOBase, but I would need to check.

File: stdlib/contextlib.pyi [hunk 2]
```diff
@@ -80,7 +81,7 @@ class ExitStack(ContextManager[ExitStack]):
     def callback(self, callback: Callable[..., Any], *args: Any, **kwds: Any) -> Callable[..., Any]: ...
     def pop_all(self: _U) -> _U: ...
     def close(self) -> None: ...
-    def __enter__(self: _U) -> _U: ...
+    def __enter__(self: Self) -> Self: ...
```
Comment (graingert) [inline stdlib/contextlib.pyi:84]: does this need a change in `__aenter__` on AsyncExitStack?

Comment (Akuli) [reply stdlib/contextlib.pyi:84]: I created #5724

**Comment by graingert** [comment]
Probably needs a scan for `__aenter__` calls

**Pull request merged**